### PR TITLE
Upgrade to Spring Boot 3.3.0

### PR DESCRIPTION
**Pull request by pacphi**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 org.gradle.parallel=true
 
 group=io.pivotal.cfenv
-version=3.1.6-SNAPSHOT
+version=3.3.0-SNAPSHOT
```
Comment (anthonydahanne) [inline gradle.properties:4]: but this version does not have to be aligned with spring boot version though?
I was by coincidence that they kinda matched.

File: build.gradle [hunk 2]
```diff
@@ -1,7 +1,7 @@
 buildscript {
     ext {
         jmockitVersion = "1.49"
-        springBootVersion = "3.2.5"
+        springBootVersion = "3.3.0"
```
Comment (anthonydahanne) [inline build.gradle:4]: what is the benefit to upgrade to latest Spring Boot?
Did they change any properties that we should update?

If not, I guess we'd want to build with the minimum version of Spring Boot 3 to ensure it's gonna work for all SP 3 versions.
I don't think it really would change anything to bump this version, so I'm not strongly against, but I'm now curious to understand what you think would be the advantage.

File: gradle.properties (hunk 1 above)
Comment (pacphi) [reply gradle.properties:4]: I will change back to 3.1.6-SNAPSHOT.

File: build.gradle (hunk 2 above)
Comment (pacphi) [reply build.gradle:4]: Main reason for update to latest Spring Boot major.minor (3.3.x as of this response) is that it keeps this library up to date in the OSS support window until May 2025; see https://spring.io/projects/spring-boot#support.

**Comment by pacphi** [comment]
I'm going to close this PR and file a new one with updated changes.

**Pull request closed without merging**